orchagent/nexthopkey.h

```diff
@@ -137,6 +144,11 @@ struct NextHopKey
     {
         return (ip_address.getV4Addr() == 0);
     }
+
+    NextHopKey ipKey() const
+    {
+        return NextHopKey(ip_address, alias);
```

Comment:
In case of EVPN/Vxlan, for the same ip & alias there can be multiple nexhtops with different vni and mac_address. If  ipKey is used to compare the nexthop objects then the results might not be valid for EVPN/Vxlan.

Reply:
This method is specifically to get the underlying IP/interface pair to check if neighorch has learned this neighbor and has programmed the basic next hop for the pair. We don't want to create e.g. a labelled next hop for the pair if the basic next hop doesn't exist yet. I'll add a comment explaining the purpose of this function in my next update.

Reply:
Done

Reply:
I don't see this addressed

Reply:
@prsunny We are using this in several places and I think it would be good to keep here, but maybe a different name would make its purpose clearer?  getNeighNhKey()?  getArpNhKey()?

Reply:
@prsunny Or it could be pushed to NeighOrch... NeighOrch::isNeighborResolved(nexthop)?

Reply:
Fixed - I've taken Ann's second solution of adding a NeighOrch method